src/internal/user_api.rs

```diff
@@ -257,9 +258,78 @@ impl UserDevice {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+struct Claims {
+    sub: String,
+    pid: usize,
+    sid: String,
+    kid: Option<usize>,
+    iat: u64,
+    exp: Option<u64>,
+}
+
+/// IronCore JWT.
+///
+/// Must be either ES256 or RS256 and have a payload similar to:
+/// ```
+/// struct Claims {
+///     sub: String,         // unique user ID
+///     pid: usize,          // project ID
+///     sid: String,         // segment ID
+///     kid: Option<String>, // service key ID
+///     iat: u64,            // issued time (seconds)
+///     exp: Option<u64>,    // expiration time (seconds)
+/// }
+/// ```
+#[derive(Clone, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
+pub struct Jwt(pub(crate) String);
+impl Jwt {
+    pub fn new(jwt: &str) -> Result<Jwt, IronOxideErr> {
+        // This `dangerous_unsafe_decode` is safe because the server will do the actual
+        // signature verification and validation. We just want to do a little initial validation
+        // to catch issues earlier.
+        let token_data = jsonwebtoken::dangerous_unsafe_decode::<Claims>(jwt)
+            .map_err(|e| IronOxideErr::ValidationError("jwt".to_string(), e.to_string()))?;
+        let alg = token_data.header.alg;
+        if alg == Algorithm::ES256 || alg == Algorithm::RS256 {
+            Ok(Jwt(jwt.to_string()))
+        } else {
+            Err(IronOxideErr::ValidationError(
+                "jwt".to_string(),
+                "Unsupported JWT algorithm. Supported algorithms: ES256 and RS256.".to_string(),
+            ))
+        }
+    }
+
+    pub fn jwt(&self) -> &str {
+        self.0.as_str()
+    }
+
+    fn to_utf8(&self) -> Vec<u8> {
+        self.0.as_bytes().to_vec()
+    }
+}
+impl std::fmt::Display for Jwt {
```

Comment:
It would be nice for either `Debug` or `Display` to show the parsed JWT. This is probably the right impl for `Display` and `Debug` should be changed.

Reply:
Done.
`Display` is just the jwt string slice, while `Debug` has the string slice, headers, and claims.